### PR TITLE
Remove Struct from everything

### PR DESCRIPTION
**Pull request by mullermp**
Removes Struct from:
* Auth
* Endpoints
* Various Hearth internal structures
* Config

This is because RBS with Struct is not well supported.

### CODE REVIEW

File: hearth/lib/hearth/structure.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def initialize(options = {})
     # Deeply converts the Struct into a hash. Structure members that
     # are `nil` are omitted from the resultant hash.
     #
-    # @return [Hash, Structure, Object]
+    # @return [Hash, Structure]
     def to_h(obj = self)
```
Comment (alextwoods) [inline hearth/lib/hearth/structure.rb:27]: Should we also implement a to_s method?

**Review by alextwoods — APPROVED** [review]
Nice!  

Minor comments.  Also - add changelogs?

File: hearth/lib/hearth/structure.rb (hunk 1 above)
Comment (mullermp) [reply hearth/lib/hearth/structure.rb:27]: Yeah, but we need a SENSITIVE members array too I think

Comment (alextwoods) [reply hearth/lib/hearth/structure.rb:27]: We currently generate a to_s method only when sensitive members exist.  

We can either:
1. Write a default to_s method on Hearth::Structure and make no changes to codegen (continue generating a custom to_s that overrides when senstive members exist).
2. Generate a to_s method for all generated Types.
3. Generate a SENSITIVE members list and then write a default to_s on Hearth::Strucutre that handles that.

My preference would be 1.  But I'm fine as long as we have both useful default printing of the types + consideration of sensitive members.

**Pull request merged**